lib/go-atscfg/astatsdotconfig.go

```diff
@@ -19,23 +19,41 @@ package atscfg
  * under the License.
  */
 
+import (
+	"github.com/apache/trafficcontrol/lib/go-tc"
+)
+
 const AstatsSeparator = "="
 const AstatsFileName = "astats.config"
 
 const ContentTypeAstatsDotConfig = ContentTypeTextASCII
 const LineCommentAstatsDotConfig = LineCommentHash
 
 func MakeAStatsDotConfig(
-	profileName string,
-	paramData map[string]string, // GetProfileParamData(tx, profile.ID, AstatsFileName)
-	toToolName string, // tm.toolname global parameter (TODO: cache itself?)
-	toURL string, // tm.url global parameter (TODO: cache itself?)
-) string {
-	hdr := GenericHeaderComment(profileName, toToolName, toURL)
+	server *tc.ServerNullable,
```

Comment:
Currently, rolling aliases are not being updated because that breaks the API client package. So either this needs to be an API-version-specific structure like `ServerV30` (or is it `ServerNullableV30`?) or we need to revisit that decision.

Reply:
ORT/atstccfg uses the objects returned by the latest TO client. I think it should probably always do that, and upgrade the objects from the previous client to them. 

This matches https://github.com/apache/trafficcontrol/blob/f42709effc973528d3a3b844ae473189e3f2321a/traffic_ops/v3-client/server.go#L196

If it needs changed, I think that needs changed before this.

Reply:
Well, that's a deprecated method of `github.com/apache/trafficcontrol/traffic_ops/client` - the unversioned client we're getting rid of in 6.0: _"Deprecated: GetServers will be removed in 6.0. Use GetServersWithHdr."_ - but it looks like `GetServersWithHdr` also returns `ServerNullable` - and that's what should change. I think there might be a breaking client change hidden there that needs to get fixed...

Reply:
Yep, looks like `github.com/apache/trafficcontrol/traffic_ops/client.GetServers` changed call signatures between 4.1.1 and 5.0; used to return `[]tc.Server`.

Reply:
I opened a PR to fix it: #5248